LoRA support in model builder (#955)
microsoft/onnxruntime-genai: src/python/py/models/README.md

python3 builder.py -i path_to_local_folder_on_disk -o path_to_output_folder -p int4 -e execution_provider -c cache_dir_to_store_temp_files
```

> ### LoRA Models

Reviewer: Since this uses `extra_options`, can you move this section to be after "Use QDQ Pattern for Quantization" and add a reference to this section in the table of contents after "Use QDQ Pattern for Quantization"?

https://github.com/microsoft/onnxruntime-genai/blob/dfbe14c39bc0486e1289332bca2003ff66a74fc7/src/python/py/models/README.md?plain=1#L5-L26

Can you also change the title's size so that it matches the title sizes that are sub-sections in the "Extra Options"?

```suggestion
#### LoRA Models
```

> This scenario is where you have a finetuned model which supports LoRA and want to get optimized model. To use adapter model you have to pass the path of the model as input

Reviewer: ```suggestion
This scenario is where you have a finetuned model with LoRA adapters and your model can be loaded in the Hugging Face style via [PEFT](https://github.com/huggingface/peft).
```

> path_to_local_folder_on_disk --> location where adapter_config.json and adapter weights are present

Reviewer: ```suggestion
- path_to_local_folder_on_disk = location where base model's weights are present
- path_to_adapter_files = location where adapter weights are present

```

```
# From wheel:

> python3 -m onnxruntime_genai.models.builder -o path_to_output_folder -p fp16 -e execution_provider -c cache_dir_to_store_temp_files --extra_options adapter_path=path_to_adapter_files

Reviewer: Can you add back the input path for the base model's weights?

```suggestion
python3 -m onnxruntime_genai.models.builder -i path_to_local_folder_on_disk -o path_to_output_folder -p fp16 -e execution_provider -c cache_dir_to_store_temp_files --extra_options adapter_path=path_to_adapter_files
```

# From source:

> python3 builder.py -o path_to_output_folder -p fp16 -e execution_provider -c cache_dir_to_store_temp_files --extra_options adapter_path=path_to_adapter_files

Reviewer: Can you add back the input path for the base model's weights?

```suggestion
python3 builder.py -i path_to_local_folder_on_disk -o path_to_output_folder -p fp16 -e execution_provider -c cache_dir_to_store_temp_files --extra_options adapter_path=path_to_adapter_files
```

python3 builder.py -i path_to_local_folder_on_disk -o path_to_output_folder -p precision -e execution_provider -c cache_dir_to_store_temp_files --extra_options use_qdq=1
```

> #### LoRA Models

Reviewer: Can you add a [reference](https://github.com/microsoft/onnxruntime-genai/pull/955#discussion_r1807486816) to this new section in the table of contents?
